crates/uv/src/commands/project/init.rs

```diff
@@ -571,6 +609,108 @@ impl InitProjectKind {
 
         Ok(())
     }
+
+    async fn init_script(
+        self,
+        script_path: &PathBuf,
+        python: Option<String>,
+        connectivity: Connectivity,
+        python_preference: PythonPreference,
+        python_downloads: PythonDownloads,
+        cache: &Cache,
+        printer: Printer,
+        no_workspace: bool,
+        no_readme: bool,
+        no_pin_python: bool,
+        package: bool,
+        native_tls: bool,
+    ) -> Result<()> {
+        if no_workspace {
+            warn_user_once!("`--no_workspace` is a no-op for Python scripts, which are standalone");
+        }
+        if no_readme {
+            warn_user_once!("`--no_readme` is a no-op for Python scripts, which are standalone");
+        }
+        if package {
+            warn_user_once!("`--package` is a no-op for Python scripts, which are standalone");
+        }
+
+        if let Some(path) = script_path.to_str() {
+            if !path.ends_with(".py") {
+                anyhow::bail!("Script name must end in .py extension");
+            }
+        }
+
+        let client_builder = BaseClientBuilder::new()
+            .connectivity(connectivity)
+            .native_tls(native_tls);
+
+        let reporter = PythonDownloadReporter::single(printer);
+
+        if script_path.try_exists()? {
```

Comment:
Using try_exists here in line with a comment on the previous PR, but should we potentially use ```fs_err::tokio::metadata``` so it's async? (Although I don't suspect it's an expensive operation)

Reply:
Yeah let's do that as a best practice.